Lib/logging/handlers.py

```diff
@@ -196,9 +196,12 @@ def shouldRollover(self, record):
         if self.stream is None:                 # delay was set...
             self.stream = self._open()
         if self.maxBytes > 0:                   # are we rolling over?
+            pos = self.stream.tell()
+            if not pos:
+                # gh-116263: Never rollover an empty file
+                return False
             msg = "%s\n" % self.format(record)
-            self.stream.seek(0, 2)  #due to non-posix-compliant Windows feature
```

Comment:
This line was removed but I don't see any discussion of what it did or why it might've existed over in the original `main` PR https://github.com/python/cpython/pull/122788.

That seems worth investigating this otherwise no-op call to make sure it isn't actually needed on modern platforms anymore?

Reply:
Fair point. Would a successful buildbot run be sufficient? That line is 22 years old at this point.

Reply:
@zooba do you know of any reason on Windows this would ever be necessary?  Context: This is on a log file being appended to that we're checking the current size of and we've got a comment in the code on an seemingly unnecessary `seek(..., SEEK_END)` before the `tell` based current position as size check.

GUESS: When this was written Python 1.5|2 was using the C file IO stack (`fwrite`, `fseek`, `ftell`, `FILE*`)... and given that logs are text files, it was _probably_ `fopen`'d in `"at"` mode(?).  Meaning I _presume_ that newline translation from \n to \r\n might have been happening on write on Windows?  Perhaps that caused an issue with `FILE*` internal record keeping of position for `ftell` vs reality? Thus an `fseek(..., SEEK_END)` before `ftell` to get a more accurate answer?  (I feel like I'm making a wild guess here and that any need for this would've been a workaround for an underlying platform bug.)

These days we obviously using CPython's own text IO stack rather than C's. I'm guessing this harmless SEEK_END is no longer necessary. Any good reasons to disagree?

This is already in `main` and didn't cause any buildbots to pipe up there (though do we even have tests that'd notice whatever discrepancy in behavior this might have addressed?).

Reply:
FWIW if my theory of this being related to newline translation (and these days: text codec encoding is a similar concept) leading to differing opinions on what current file position is (number of characters passed to write vs number of encoded bytes?) the net effect would be doing log rotation probably a bit later than strictly configured, but within the same rough size magnitude, so I doubt any users would really care.

Reply:
Yeah, I don't have any idea. The newline translation theory is as good as any other. I have a vague recollection of "you should always `seek(0, CURRENT)` to get the current location" rather than an `ftell` equivalent, but I don't remember why. And that might be from the Visual Basic 6 days!

As far as I'm aware, our own `.tell()` is reliable, regardless of newline processing. And I agree with your analysis of the effect in this case, so it's not a big deal either way.

Reply:
Okay, lets stick with the removal as has already landed in `main`.  One less line of mystery code! :)